src/coreclr/jit/compiler.cpp

```diff
@@ -2297,6 +2297,13 @@ void Compiler::compSetProcessor()
             codeGen->GetEmitter()->SetUseEvexEncoding(true);
             // TODO-XArch-AVX512 : Revisit other flags to be set once avx512 instructions are added.
         }
+        if (canUseRex2Encoding() || DoJitStressRex2Encoding())
```

Comment:
Is the `DoJitStressRex2Encoding()` check needed here? We notably don't have the equivalent for the `canUseEvexEncoding()` path even though a `DoJitStressEvexEncoding()` check exists.

Reply:
we will eventually turn this part into in the APX-EVEX PR:
```
if (canUseApxEncoding)
{
   codeGen->GetEmitter()->SetUseRex2Encoding(true);
   codeGen->GetEmitter()->SetUsePromotedEVEXEncoding(true);
}
```

the code here in this branch is a bit outdated, I will mirror the changes to this PR.